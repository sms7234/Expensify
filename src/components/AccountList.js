import React from  'react';
import {connect} from 'react-redux';
import AccountListItem from './AccountListItem';
import selectAccounts from '../selectors/accounts';


export const AccountList = (props) => (
  <div className="content-container">
    <div className="list-header">
      <div className="show-for-mobile">Accounts</div>
      <div className="show-for-desktop">Accounts</div>
      <div className="show-for-desktop">Description</div>
    </div>
    <div className="list-body">
    {
      props.accounts.length === 0 ? (
        <div className="list-item--message">
          <span>No accounts to show</span>
        </div>
      ): (
        props.accounts.map((item) => {return <AccountListItem key={item.id}{...item}/>;})
      )
    }
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    accounts: selectAccounts(state.accounts, state.filters)
  };
};

export default connect(mapStateToProps)(AccountList);
