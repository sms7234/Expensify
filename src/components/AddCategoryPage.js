import React from 'react';
import CategoryForm from './CategoryForm';
import {connect} from 'react-redux';
import {startAddCategory} from '../actions/categories';

export class AddCategoryPage extends React.Component{
  onSubmit=(category) => {
    this.props.startAddCategory(category);
    this.props.history.push('/categories');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Category </h1>
          </div>
        </div>
        <div className="content-container">
          <CategoryForm onSubmit={this.onSubmit} />
        </div>
      </div>
    );
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddCategory:(category) => dispatch(startAddCategory(category))
});

export default connect(undefined, mapDispatchToProps)(AddCategoryPage);
