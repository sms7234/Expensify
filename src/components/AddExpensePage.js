import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect} from 'react-redux';
import startAddExpense from '../actions/expenses';

export class AddExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push('/expenses');
  };
  render(){
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title"> Add Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit = {this.onSubmit}
            categoryList = {this.props.categoryList}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps=(state) => {
  return {
    categoryList: state.categories.map((item)=>{return({label: item.category, value: item.category})})
  };
};

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(mapStateToProps, mapDispatchToProps)(AddExpensePage);
