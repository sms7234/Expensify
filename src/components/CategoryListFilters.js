import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter} from '../actions/filters';

export class CategoryListFilters extends React.Component {
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  render () {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="search categories"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(CategoryListFilters);
