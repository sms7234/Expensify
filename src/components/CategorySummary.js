import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export const CategorySummary = (props) => {

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Total number of categories: <span>{props.categories.length}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/createCategory">Add Category</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    categories: state.categories
  };
};

export default connect(mapStateToProps)(CategorySummary);
