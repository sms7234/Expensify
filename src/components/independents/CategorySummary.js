import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

export const CategorySummary = ({count}) => {

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title"> Category List </h1>
        <h2 className="page-header__subtitle">Total number of categories: <span>{count}</span></h2>
        <div className="page-header__actions">
          <Link className="button" to="/createCategory">Add Category</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    count: state.categories.length
  };
};

export default connect(mapStateToProps)(CategorySummary);
