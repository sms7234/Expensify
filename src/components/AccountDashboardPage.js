import React from 'react';
import AccountList from './AccountList';
import AccountSummary from './AccountSummary';
import AccountListFilters from './AccountListFilters';

const AccountDashboardPage = () => (
  <div>
    <AccountSummary />
    <AccountListFilters />
    <AccountList />
  </div>
)

export default AccountDashboardPage;
