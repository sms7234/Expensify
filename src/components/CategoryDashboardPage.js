import React from 'react';
import CategoryList from './CategoryList';
import CategoryListFilters from './CategoryListFilters';
import CategorySummary from './CategorySummary';

const CategoryDashboardPage = () => (
  <div>
    <CategorySummary />
    <CategoryListFilters />
    <CategoryList />
  </div>
)

export default CategoryDashboardPage;
