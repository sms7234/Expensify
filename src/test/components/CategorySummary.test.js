import React from 'react';
import {shallow} from 'enzyme';
import {CategorySummary} from '../../components/CategorySummary';

test('should correctly render category summary with no expenses',() => {
  const wrapper = shallow(<CategorySummary count={0} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render category summary with multiple expenses',() => {
  const wrapper = shallow(<CategorySummary count={47} />);
  expect(wrapper).toMatchSnapshot();
});
