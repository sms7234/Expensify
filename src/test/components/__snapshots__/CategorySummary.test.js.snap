// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render category summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Category List 
    </h1>
    <h2
      className="page-header__subtitle"
    >
      Total number of categories: 
      <span>
        47
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/createCategory"
      >
        Add Category
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render category summary with no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Category List 
    </h1>
    <h2
      className="page-header__subtitle"
    >
      Total number of categories: 
      <span>
        0
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/createCategory"
      >
        Add Category
      </Link>
    </div>
  </div>
</div>
`;
